feat(boilerplate): add empty state component to demoPodcastListScreen

kateinkim opened the pull request:
## Please verify the following:

- [ ] `yarn ci:test` **jest** tests pass with new tests, if relevant
- [ ] `README.md` has been updated with your changes, if relevant

## Describe your PR

Per issue #2079 `ListEmptyComponent` prop is added to show an empty state component when the podcasts list is somehow not loaded. Note that the design is temporary.


## Photos/Videos
|Empty state component| Overall flow|
|-|-|
|<img width="313" alt="CleanShot 2022-09-01 at 09 21 51@2x" src="https://user-images.githubusercontent.com/53795920/187959427-a5aabf91-0dc5-4039-8462-b6bca2fb3dc1.png">|![CleanShot 2022-09-01 at 09 26 11](https://user-images.githubusercontent.com/53795920/187959497-06d16994-5556-49c1-aa0e-2de8465d08ef.gif)|

## Issue

A. As you can see here when the app is loaded for the first time the empty state component will show up for a few seconds but not on reload.
B. When there are no favorite items the component will also show up - _working on fixing this_
 
|Issue A|Issue B|
|-|-|
|![CleanShot 2022-09-01 at 09 21 06](https://user-images.githubusercontent.com/53795920/187959368-25c50382-43a9-415b-88f8-2b301d744478.gif)|<img width="309" alt="CleanShot 2022-09-01 at 10 01 07@2x" src="https://user-images.githubusercontent.com/53795920/187960504-21236fef-6928-4bd4-a3e2-f2954940bc29.png">|

kateinkim commented:
The issue B which was addressed previously is fixed

<img src="https://user-images.githubusercontent.com/53795920/188015024-ff2fcf49-d65d-4b99-87ae-28af3be27943.gif" width="250">

kateinkim commented on boilerplate/app/screens/DemoPodcastListScreen.tsx:
```diff
@@ -57,6 +58,37 @@ export const DemoPodcastListScreen = observer(function DemoPodcastListScreen(
+                </View>
+              ) : (
+                <View>
+                  <Text preset="subheading" style={$subheading}>
+                    You don't have any favorites yet
+                  </Text>
+                  <Text>Please click the heart icon on each list to add it to your favorites.</Text>
+                </View>
```
These texts are not added to i18n since they will probably be changed when the actual design is ready.

kateinkim commented on boilerplate/app/screens/DemoPodcastListScreen.tsx:
```diff
@@ -57,6 +58,37 @@ export const DemoPodcastListScreen = observer(function DemoPodcastListScreen(
+            <View style={$container}>
+              {!episodeStore.favoritesOnly ? (
+                <View>
+                  <Text preset="subheading" style={$subheading}>
+                    So empty... so sad
+                  </Text>
+                  <Text>No data found yet. You can click the button or reload the app.</Text>
+                </View>
+              ) : (
+                <View>
+                  <Text preset="subheading" style={$subheading}>
+                    You don't have any favorites yet
+                  </Text>
+                  <Text>Please click the heart icon on each list to add it to your favorites.</Text>
```
These texts are not added to i18n yet since they will probably be changed when the actual design is ready.

kateinkim replied:
I'm still waiting for the actual design but would you take a look at this, especially the `Issue A` that I mentioned? I'm not sure if that's an ok behavior or not. @robinheinze

robinheinze commented:
@kateinkim For Issue A, I think it would make more sense to have a loading state. So when we start to fetch episodes, set a loading flag to true, then to false once we've finished fetching. Then in the empty state, we show the loading spinner if that loading flag is true, and otherwise we can show the empty state. So it only shows if we tried to load data and didn't get any. Does that make sense?

kateinkim commented:
Yes! It totally makes sense. 👍🏼

kateinkim commented:
Fixed Issue A

https://user-images.githubusercontent.com/53795920/188236069-079bd7ff-7b3e-41d6-8ecb-7f8f90663a05.mp4

kateinkim commented:
1. Updated the empty state component design according to the actual design
2. Also added the paddingTop on each screen to match each other per design spacing guide as well.

|Empty State|No favorites|
|-|-|
|<img width="320" alt="CleanShot 2022-09-07 at 16 22 35@2x" src="https://user-images.githubusercontent.com/53795920/189006918-b0623f4a-ff64-483d-befb-8af749b06cfe.png">|<img width="331" alt="CleanShot 2022-09-07 at 16 21 56@2x" src="https://user-images.githubusercontent.com/53795920/189006945-6b85380e-27b5-448e-b6c0-98b848bb9562.png">|
<img width="313" alt="CleanShot 2022-09-08 at 08 18 09@2x" src="https://user-images.githubusercontent.com/53795920/189146339-cc0b7302-5b96-4008-9241-5f8b7449d647.png">



|

kateinkim commented:
1. Updated the empty state component according to the design.
2. Added more spacing on the top of the each screens.

|No data empty state|No favorites empty state|
|-|-|
|

mazenchami commented on boilerplate/app/i18n/ar.ts:
```diff
@@ -97,6 +97,14 @@ const ar: Translations = {
+    noFavoritesEmptyState: {
+      title: "This looks a bit empty",
+      message: "No favorites have been added yet. Tap the heart on an episode to add it to your favorites!",
+    }
```
@kateinkim done here: https://github.com/infinitered/ignite/pull/2108/commits/cfe69236da5f327fbc9460b3f4d7da8442d8afee

mazenchami commented on boilerplate/app/screens/DemoPodcastListScreen.tsx:
```diff
@@ -252,4 +293,34 @@ const $metadata: TextStyle = {
+const $sadFaceContainer: ViewStyle = {
+  alignItems: "center",
+  paddingTop: spacing.huge,
+  paddingBottom: spacing.extraSmall,}
```
[nits] the `}` should be on a new line

kateinkim replied:
got it! Good catch!

jamonholmgren reviewed (approved):
Good to go!

robinheinze reviewed (approved):
Looks great!
